include/data_structures/data_types.h

```diff
@@ -34,6 +34,13 @@ inline double zero() {
   return 0.;
 }
 
+//! Position type
+//! None: No position is specified
+//! Corner: Nodes at boundary corners
+//! Edge: Nodes along boundary edges
+//! Face: Nodes on boundary faces
+enum class Position { None, Corner, Edge, Face };
```

Comment:
I think this is only ok for nodes. This will be refactored/removed once we implement a more clever scheme for B-spline anyway. Were you thinking something like
```
enum class NodePosition { None, Corner, Edge, Face };
```

Reply:
Okay! Let's keep it for now as it is. Let's change it as we go towards B-Spline